src/bosh-director/lib/bosh/director/instance_updater.rb

```diff
@@ -72,6 +72,23 @@ def needs_recreate?(instance_plan)
       false
     end
 
+    private_class_method def self.rendered_templates_persister
+      RenderedTemplatesPersister.new(
+        App.instance.blobstores.blobstore, Config.logger
+      )
+    end
+
+    private_class_method def self.disk_manager
+      DiskManager.new(Config.logger)
+    end
+
+    private_class_method def self.vm_creator(dns_encoder, template_blob_cache,
+                                             link_provider_intents)
+      VmCreator.new(Config.logger, template_blob_cache,
+                    dns_encoder, AgentBroadcaster.new,
+                    link_provider_intents)
+    end
+
```

Comment:
Extracting logic from a class method constructor (like `new_instance_updater`) doesn't appear to be too common. I'll leave the choice of refactoring it to be more consistent vs readability to you here.

My preference personally would be to keep `new_instance_updater` the way it was (+any rubocop lints that came up).